pkg/sql/sem/builtins/geo_builtins.go

```diff
@@ -2815,6 +2917,108 @@ The requested number of points must be not larger than 65336.`,
 			tree.VolatilityImmutable,
 		),
 	),
+	"st_ymin": makeBuiltin(
+		defProps(),
+		geometryOverload1(
+			func(ctx *tree.EvalContext, g *tree.DGeometry) (tree.Datum, error) {
+				t, err := g.Geometry.AsGeomT()
+				if err != nil {
+					return nil, err
+				}
+				switch t := t.(type) {
+				case geom.T:
+					if t.Empty() {
+						return tree.DNull, nil
+					}
+					if t.Stride() < 2 {
+						return tree.NewDFloat(0.00), nil
+					}
+					bounds := t.Bounds()
+					return tree.NewDFloat(tree.DFloat(bounds.Min(1))), nil
+				}
+				// Ideally we should return NULL here, but following PostGIS on this.
+				return nil, errors.Newf("argument to st_ymin() must be a valid geometry")
+			},
+			types.Float,
+			infoBuilder{
+				info: "Returns the min(Y) coordinate of a geometry.",
+			},
+			tree.VolatilityImmutable,
+		),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"box2d", types.Box2D},
+			},
+			ReturnType: tree.FixedReturnType(types.Float),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				bbox := tree.MustBeDBox2D(args[0]).CartesianBoundingBox
+				t := bbox.ToGeomT(0 /* SRID */)
+				if t.Empty() {
+					return tree.DNull, nil
+				}
+				if t.Stride() < 2 {
+					return tree.NewDFloat(0.00), nil
+				}
+				bounds := t.Bounds()
+				return tree.NewDFloat(tree.DFloat(bounds.Min(1))), nil
+			},
+			Info: infoBuilder{
+				info: "Returns the min(Y) coordinate of a box2d.",
+			}.String(),
+			Volatility: tree.VolatilityImmutable,
+		},
+	),
+	"st_ymax": makeBuiltin(
+		defProps(),
+		geometryOverload1(
+			func(ctx *tree.EvalContext, g *tree.DGeometry) (tree.Datum, error) {
+				t, err := g.Geometry.AsGeomT()
```

Comment:
you should be able to infer the geometry bounding box from `g.Geometry.BoundingBox`. if it is nil, it is empty.
this should work for X and Y

```
bbox := g.Geometry.BoundingBox
if bbox == nil {
   return tree.DNull, nil
}
return tree.NewDFloat(bbox.MinX), nil
```